File: go/15_enums.go
package main

import "fmt"

type Status int

const (
	Pending Status = iota
	Running
	Done
	Failed
)

func (s Status) String() string {
	return [...]string{"pending", "running", "done", "failed"}[s]
}

func main() {
	status := Running
	fmt.Println(status)
	fmt.Println(status.String())

	if status == Running {
		fmt.Println("still working...")
	}
}

File: rust/15_enums.rs
#[derive(Debug, PartialEq)]
enum Status {
    Pending,
    Running,
    Done,
    Failed,
}

fn main() {
    let status = Status::Running;
    println!("{:?}", status);

    match status {
        Status::Pending => println!("waiting"),
        Status::Running => println!("still working..."),
        Status::Done => println!("finished"),
        Status::Failed => println!("error"),
    }
}
